### PR TITLE
Event creation

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

File: moderator/moderate/views.py [hunk 1]
```diff
@@ -56,7 +56,60 @@ def archive(request):
 
 
 @login_required(login_url='/')
-def event(request, e_slug, q_id=None):
+def edit_event(request, slug=None):
+    """Creates a new event."""
+    user = request.user
+    query_args = {
+        'slug': slug,
```
Comment (johngian) [inline moderator/moderate/views.py]: If ``slug=None``, ``slug`` shouldn't be part of ``query_args``.

File: moderator/moderate/views.py [hunk 2]
```diff
@@ -56,7 +56,60 @@ def archive(request):
...
+    # Allow superusers to edit all events
+    if user.is_superuser:
+        del query_args['created_by']
+
+    event = get_object_or_404(Event, **query_args) if slug else None
+    event_form = EventForm(request.POST or None, instance=event)
+
+    if event_form.is_valid():
+        event = event_form.save(commit=False)
+        event.created_by = user
```
Comment (johngian) [inline moderator/moderate/views.py]: Why do you want to override ``created_by`` ?

File: moderator/moderate/views.py [hunk 3]
```diff
@@ -56,7 +56,60 @@ def archive(request):
...
+
+        if slug:
+            msg = 'Event successfully edited.'
+        else:
+            msg = 'Event successfully created.'
+        messages.success(request, msg)
+        return redirect(reverse('main'))
+
+    return render(request, 'create_event.jinja',
+                  {'slug': event.slug if event else None,
```
Comment (johngian) [inline moderator/moderate/views.py]: nit: Can you please move this to a var eg. ``ctx`` ?

**Comment by johngian** [comment]
Please fix failing tests. r+wc

File: moderator/moderate/views.py (hunk 2 above)
Comment (akatsoulas) [reply moderator/moderate/views.py]: This is a new field and it's not part of the event form.

**Pull request merged**